src/native/eventpipe/ds-profiler-protocol.c

```diff
@@ -83,6 +97,71 @@ ds_attach_profiler_command_payload_free (DiagnosticsAttachProfilerCommandPayload
 	ep_rt_object_free (payload);
 }
 
+static
+bool
+profiler_protocol_helper_attach_profiler (
+	DiagnosticsIpcMessage *message,
+	DiagnosticsIpcStream *stream)
+{
+	EP_ASSERT (message != NULL);
+	EP_ASSERT (stream != NULL);
+
+	if (!stream)
+		return false;
+
+    bool result = false;
+    DiagnosticsAttachProfilerCommandPayload *payload = NULL;
+
+    if (!ep_rt_is_running ()) {
+        ds_ipc_message_send_error (stream, DS_IPC_E_NOT_YET_AVAILABLE);
+        ds_ipc_stream_free (stream);
+        ep_raise_error ();
```

Comment:
I'm not sure if this is the double free you are hitting, but this will lead to a double free.

`ep_raise_error ()` => `goto ep_on_error` which ends with `ep_exit_error_handler ()` => `goto ep_on_exit` which _also_ calls `ds_ipc_stream_free (stream)`.

Reply:
I think this was it, the failing test was using the profiler attach command in reverse startup mode and verifying it failed, which would have hit this code path.